### PR TITLE
AWS foundational security Lambda.5 does not work as in Security Hub

### LINKED ISSUE #374: AWS foundational security Lambda.5 does not work as in Security Hub
**Describe the bug**
AWS https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-standards-fsbp-controls.html#fsbp-lambda-5 says that a Lambda function should operate in more than one availability zone (AZ).

What the AWS documentation is not saying that this should only be checked *if* a lambda function has VPC attachment at all.
The AWS Security HUB ignored Lambda functions, which are *not* attached to a VPC.

Security wise this is the right behaviour, because this check wants the ensure high availability for Lambda functions.
If a Lambda is outside from an VPC, then AWS ensures high availability, so the check should pass.

*If* a Lambda is attached to a VPC, it is not enough to check whether it is attached to more than one subnet. It must be checked, that the subnets are located in different AZ.

Therefor the query for foundational_security_lambda_5 should be:

```sql

select
  -- Required Columns
  arn as resource,
  case
    when vpc_id is null then 'ok'
    else case    
      when 
      (select count(distinct availability_zone_id) from aws_vpc_subnet 
        where 
          subnet_id in (select jsonb_array_elements_text(vpc_subnet_ids) )  
      ) >= 2
      then 'ok'
    else 'alarm'
    end
  end as status,
  case
    when vpc_subnet_ids is null then title || ' is not in VPC.'
    else title || ' has ' || jsonb_array_length(vpc_subnet_ids) || ' availability zone(s).'
  end as reason,
  -- Additional Dimensions
  region,
  account_id
from
  "aws_lambda_function";
```
 
 This is the Security hub behaviour with the Lambda described in "to reproduce"
 
Lambda-5-security-hub

This is the old, wrong steampipe behaviour:

 
Lambda-5-steampipe-bug

**Steampipe version (`steampipe -v`)**
0.13.6

**Plugin version (`steampipe plugin list`)**
0.57.0

**To reproduce**
1) create Lambda with name (name)
A) Lambda _without_ VPC attachment (amplify-login-custom-message-73508ae0)
B) Lambda with VPC attachment in one subnet (InVPConeSubnet)
C) Lambda with VPC attachment in two subnet in same AZ (InVPCtwosameSubnet)
D) Lambda with VPC attachment in two subnet in different AZ (InVPCtwoSubnet)

**Expected behavior**

Lambda-5-steampipe-fixed

**Additional context**

### PR DESCRIPTION
**Pull request by megaproaktiv**
### Checklist
Fixes #374 

Also changed status to "skipped", which is more accurate

### CODE REVIEW

File: query/lambda/lambda_function_multiple_az_configured.sql [hunk 1]
```diff
@@ -1,10 +1,18 @@
+
```
Comment (cbruno10) [inline query/lambda/lambda_function_multiple_az_configured.sql:1]: @megaproaktiv Can you please remove any empty newlines, like here, and also below at the bottom of the file?

File: query/lambda/lambda_function_multiple_az_configured.sql [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
+      (select count(distinct availability_zone_id) from aws_vpc_subnet 
+        where 
+          subnet_id in (select jsonb_array_elements_text(vpc_subnet_ids) )  
+      ) >= 2
+      then 'ok'
     else 'alarm'
+    end
   end as status,
   case
     when vpc_subnet_ids is null then title || ' is not in VPC.'
```
Comment (cbruno10) [inline query/lambda/lambda_function_multiple_az_configured.sql:18]: @megaproaktiv Can you please also update the case here for the `reason` to match the case above?

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@megaproaktiv Thanks for opening a new PR with additional changes!

I've left a few suggestions here, but can you also please take a look at some of my questions/suggestions from your original PR (https://github.com/turbot/steampipe-mod-aws-compliance/pull/371#pullrequestreview-952231673)? I believe most of these are still relevant in this PR as well.

File: query/lambda/lambda_function_multiple_az_configured.sql (hunk 1 above)
Comment (megaproaktiv) [reply query/lambda/lambda_function_multiple_az_configured.sql:1]: sure!

File: query/lambda/lambda_function_multiple_az_configured.sql (hunk 2 above)
Comment (megaproaktiv) [reply query/lambda/lambda_function_multiple_az_configured.sql:17]: You mean the indentation? Sure!

Comment (cbruno10) [reply query/lambda/lambda_function_multiple_az_configured.sql:17]: @megaproaktiv Not necessarily the indentation, but when evaluating the `state`, you updated the check to be `when vpc_id is null then 'skip'`, which seems like a good change. We should perform the same check when evaluating the `reason` as well, so `when vpc_subnet_ids is null then title || ' is not in VPC.'` should probably become `when vpc_id is null then title || ' is not in VPC.'`

**Comment by cbruno10** [comment]
@megaproaktiv Thanks for creating the 2 Lambda query issues and raising the PRs, we appreciate the reports!

We will not be accepting this PR at this time, due to the number of other non-standard changes included in the branch, e.g., extra files, wrapping table names in quotes, etc., but we have used your changes as inspiration as part of https://github.com/turbot/steampipe-mod-aws-compliance/pull/392. These changes will be included as part of v0.33, which will be released next week.

If you have any other questions or issues, please let us know!

**Pull request closed without merging**